src/Unit.hs

```diff
@@ -45,15 +51,30 @@ class Applicative u => Unit (dim :: * -> *) u | u -> dim where
 
   suffix :: Const ShowS (u a)
 
-unitary :: forall u a b dim . (Unit dim u, Fractional a, Fractional b) => Iso (u a) (u b) a b
-unitary = iso ((* getConst (factor @_ @u)) . prj) (pure . (/ getConst (factor @_ @u)))
+instance Unit I I where
+  suffix = Const (showChar '1')
+
+instance Unit Identity Identity where
+  suffix = Const (showChar '1')
 
 convert :: forall u u' a dim . (Unit dim u, Unit dim u', Fractional a) => u a -> u' a
 convert = pure . (/ getConst (factor @_ @u')) . (* getConst (factor @_ @u)) . prj
 
+convertFrom :: (Unit dim u, Unit dim u', Fractional a) => (forall a . a -> u a) -> u a -> u' a
+convertFrom _ = convert
+
+convertTo :: (Unit dim u, Unit dim u', Fractional a) => (forall a . a -> u' a) -> u a -> u' a
+convertTo _ = convert
+
 converting :: forall u u' a b dim . (Unit dim u, Unit dim u', Fractional a, Fractional b) => Iso (u a) (u b) (u' a) (u' b)
 converting = iso convert convert
 
+convertingFrom :: (Unit dim u, Unit dim u', Fractional a, Fractional b) => (forall a . a -> u a) -> Iso (u a) (u b) (u' a) (u' b)
+convertingFrom _ = converting
+
+convertingTo :: (Unit dim u, Unit dim u', Fractional a, Fractional b) => (forall a . a -> u' a) -> Iso (u a) (u b) (u' a) (u' b)
+convertingTo _ = converting
```

Comment:
This is super handy since applying the type parameter for the destination unit (usually the one you need to specify) is awkward.